### PR TITLE
Implement attribute set with fallbacks

### PR DESCRIPTION
**Pull request by arothuis**
This implements #55 in Profile, using the ConfigurableAttributeSetFactory from SURFnet/Stepup-saml-bundle#51.

As shown in the image below, an undefined attribute will use its urn as a name in the My Services table (in this case I removed the definition for the "surfconext status" for testing purposes). It will not throw an exception.
screen shot 2016-10-19 at 14 56 03

### CODE REVIEW

**Comment by arothuis** [comment]
@rjkip

File: src/OpenConext/Profile/Entity/AuthenticatedUser.php [hunk 1]
```diff
@@ -34,7 +34,7 @@
     private $nameId;
 
     /**
-     * @var AttributeSet
+     * @var AttributeSetInterface
```
Comment (rjkip) [inline src/OpenConext/Profile/Entity/AuthenticatedUser.php]: Come to think of it, this interface, returned from the factory, is actually a BC break, innit?

File: src/OpenConext/ProfileBundle/Resources/views/MyServices/overview.html.twig [hunk 2]
```diff
@@ -76,7 +76,13 @@
                                     {% endif %}
                                     <tr>
                                         <td title="{{ attributeUrn }}">
-                                            <strong>{{ ('profile.saml.attributes.' ~ attribute.attributeDefinition.name)|trans({}, 'saml') }}</strong>
+                                            {% set translatedAttribute = ('profile.saml.attributes.' ~ attribute.attributeDefinition.name)|trans({}, 'saml') %}
+
+                                            {% if translatedAttribute == 'profile.saml.attributes.' ~ attribute.attributeDefinition.name %}
```
Comment (rjkip) [inline src/OpenConext/ProfileBundle/Resources/views/MyServices/overview.html.twig]: So... why did we prefix the attributes in the SAML translation domain with `profile.saml.attributes.`?

Comment (arothuis) [inline src/OpenConext/ProfileBundle/Resources/views/MyServices/overview.html.twig]: I think we wanted to make them look like translation keys in order to distinguish the translation key from the raw URN.
I might rename this to `translatedAttributeName`.

Comment (rjkip) [inline src/OpenConext/ProfileBundle/Resources/views/MyServices/overview.html.twig]: Meh, I'll allow it :hammer:

**Pull request merged**